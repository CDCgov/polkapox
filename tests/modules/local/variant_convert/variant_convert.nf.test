nextflow_process {

    name "Test Process VARIANT_CONVERT"
    script "modules/local/variant_convert/variant_convert.nf"
    process "VARIANT_CONVERT"

    test("Variant Convert Test") {

        when {
            params {
            }

            process {
                """
                input[0] = Channel.of(
                    [
                        meta: [id: 'test_sub'],
                        tsv: file("${projectDir}/tests/nf-test-data/test_sub.ivar.tsv", checkIfExists: true)
                    ]
                )
                input[1] = [af_cutoff: '0.01']
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
