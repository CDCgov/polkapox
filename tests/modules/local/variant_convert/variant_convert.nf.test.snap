{
    "Variant Convert Test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "vcf": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:12:38.15682358"
    },
    "Should convert ivar TSV to VCF without error": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "vcf": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-08T13:42:27.638373472"
    }
}